src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -525,14 +616,18 @@ class ALFASimResultMetadata:
         Map "base time steps" to the application version used in the simulation.
     """
 
-    profiles = attr.ib(validator=attr.validators.instance_of(dict))
-    trends = attr.ib(validator=attr.validators.instance_of(dict))
-    time_sets = attr.ib(validator=attr.validators.instance_of(list))
-    time_sets_unit = attr.ib(validator=attr.validators.instance_of(str))
-    time_steps_boundaries = attr.ib(validator=attr.validators.instance_of(tuple))
-    time_set_info = attr.ib(validator=attr.validators.instance_of(dict))
-    result_directory = attr.ib(validator=attr.validators.instance_of(Path))
-    app_version_info = attr.ib(validator=attr.validators.instance_of(dict))
+    profiles: dict[OutputKeyType, ProfileMetaItem]
+    trends: dict[OutputKeyType, TrendMetaItem]
+    time_sets: list[SourceTimeSetKeyType]
+    time_sets_unit: str
+    time_steps_boundaries: tuple[
+        tuple[TimeStepIndex, TimeStepIndex], tuple[TimeStepIndex, TimeStepIndex]
+    ]
+    time_set_info: dict[
+        Literal["profiles", "trends"], dict[TimeStepIndex, TimeSetInfoItem]
+    ]  # TODO: verify if keys are TimeStep.
```

Comment:
TODO

Reply:
- [ ] Remove TODO

I would ask @prusse-martin on this one but he is on vacation right now. I will leave this as `TimeStepIndex` for now but will ask him to do a post-mortem review on this and in the `Trend/ProfileMetaItem` definitions, then any changes can be addressed in a follow-up PR.